src/main/kotlin/lotto/domain/LottoRank.kt

```diff
@@ -1,18 +1,25 @@
 package lotto.domain
 
 enum class LottoRank(
-    val matchCount: Int?,
+    val countOfMatch: Int?,
     val price: Int
 ) {
-    NONE(null, 0),
-    FOURTH(3, 5000),
-    THIRD(4, 50000),
-    SECOND(5, 1500000),
-    FIRST(6, 2000000000);
+    FIRST(6, 2_000_000_000),
+    SECOND(5, 30_000_000),
+    THIRD(5, 1_500_000),
+    FOURTH(4, 50_000),
+    FIFTH(3, 5_000),
+    MISS(0, 0);
 
     companion object {
-        fun getLottoRankByMatchCount(count: Int): LottoRank {
-            return LottoRank.values().find { it.matchCount == count } ?: NONE
+        fun of(countOfMatch: Int, matchBonus: Boolean): LottoRank {
+            if (SECOND.countOfMatch == countOfMatch && matchBonus) {
+                return SECOND
+            }
+            if (SECOND.countOfMatch == countOfMatch && !matchBonus) {
+                return THIRD
+            }
+            return values().find { it.countOfMatch == countOfMatch } ?: MISS
         }
```

Comment:
when 절을 활용해볼 수 있겠네요!
```suggestion
        fun of(countOfMatch: Int, matchBonus: Boolean): LottoRank {
            return when {
                SECOND.countOfMatch == countOfMatch && matchBonus -> SECOND
                SECOND.countOfMatch == countOfMatch && !matchBonus) -> THIRD
                else -> values().find { it.countOfMatch == countOfMatch } ?: MISS
        }
```